cli/lsp/diagnostics.rs

```diff
@@ -290,6 +296,56 @@ struct ChannelUpdateMessage {
   batch_index: Option<usize>,
 }
 
+#[derive(Clone, Debug)]
+struct SpecifierState {
+  has_no_cache_diagnostic: bool,
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct DiagnosticsState {
+  specifiers: HashMap<ModuleSpecifier, (Option<i32>, SpecifierState)>,
```

Comment:
Do we ever remove entries from this hash map?

Reply:
No, but we probably can in `did_close`

Reply:
Can you update the PR to do that?